src/access/AccessControl.cpp

```diff
@@ -165,33 +165,48 @@ AccessControl::Delegate AccessControl::mDefaultDelegate;
 
 CHIP_ERROR AccessControl::Init()
 {
+    VerifyOrReturnError(!mIsInitialized, CHIP_ERROR_INCORRECT_STATE);
+
     ChipLogProgress(DataManagement, "AccessControl: initializing");
-    return mDelegate.Init();
+    CHIP_ERROR retval = mDelegate->Init();
```

Comment:
nit: A comment of 'mDelegate by design can never be null' may help here. My thought when reading the code was "what if delegate is null" but it took a while later to see that case never happens.